### DISCUSSION TITLE
Authelia Authentication via URL

### DISCUSSION THREAD

**Discussion by g0r3** [authelia/authelia#3958] [Q&A] [answered]
Hi,

sorry for using Github to ask a question but I couldn't find a real authelia support 'forum' where to ask.

I am using Authelia for protecting some of my self hostet services and it really works nicely. But one thing I am missing and I was not able to find in the documentation is the best way on how to deal with applications that need to authenticate but cannot use Authelia authentication form.

Can I configure Authelia to use basic auth via url like `https://user:password@subdomain.domain.tld` for all services which don't have the possibility to use the login form, while at the same time keep the auth page active?

**Comment by james-d-elliott** [comment, marked as answer]
As far as I am aware that's not possible with web standards, credentials sent via the browser basic auth shortcut are only sent if the server responds with a WWW-Authenticate header, and you should not respond with that and redirect (as the former shows the username password prompt). I may however be mistaken.

You can however have those services encode a username and password into the `Proxy-Authorization` or `Authorization` header with the basic scheme though (you'd need to have your proxy rewrite the `Authorization`  header to `Proxy-Authorization` if you use the `Authorization` header). This will suffice for 1FA. If you're unsure of how to do the encoding I can show you.
